components/CheckDocumentation/vacuum/UnnecessaryTableGrowth.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+
+import {
+  CheckDocs,
+  CheckGuidanceProps,
+  CheckTriggerProps,
+} from "../../../util/checks";
+import { formatBytes } from "../../../util/format";
+
+const UnnecessaryTableGrowthTrigger: React.FunctionComponent<
+  CheckTriggerProps
+> = ({ config }) => {
+  return (
+    <p>
+      Detects when a table has experienced unnecessary growth due to
+      insufficient VACUUM. It analyzes the table statistic from the past seven
+      days and calculates the amount of growth that could have been avoided if
+      VACUUM had been performed more frequently. Creates an issue when the
+      unnecessary growth exceeds{" "}
+      <code>{formatBytes(Number(config.settings["notify_bytes"]))}</code>, as
+      well as <code>{config.settings["notify_pct"]}%</code> of the current size.
+      Resolves once the unnecessary growth decreases due to changes in
+      autovacuum settings or table usage.
+    </p>
+  );
+};
+
+const UnnecessaryTableGrowthGuidance: React.FunctionComponent<
+  CheckGuidanceProps
+> = () => {
+  // UnnecessaryTableGrowth uses a similar pattern as MissingIndex.
+  // The docs doesn't have the solution section as it's in dynamic guidance generated in app side.
+  return (
+    <div>
+      <h4>Impact</h4>
+      <p>
+        In Postgres, any DELETEs or UPDATEs create dead rows that can be
+        VACUUMed to become reusable space for future INSERTs or UPDATEs.
+        Autovacuum is designed to automatically perform VACUUM regularly to
+        efficiently reclaim reusable space created by removing dead rows. This
+        helps ensure that INSERTs and UPDATEs can use existing space instead of
+        claiming new space. In this way, the table can avoid unnecessary growth.
+      </p>
+      <p>
+        However, if autovacuum is not performed with the appropriate frequency
+        or at the right time, dead rows may not be reclaimed. This can cause
+        unnecessary table growth, which could have been avoided if dead rows
+        were reclaimed. This typically result in table bloat where the physical
+        size of the table on disk is larger than its actual data size. Table
+        bloat is problematic for several reasons:
+      </p>
+      <ul>
+        <li>
+          <h5>Slower queries</h5>
+          <p>
+            When a table is bloated, it takes more time to read and write data
+            from the disk, which can slow down queries that access the table.
+          </p>
+        </li>
+        <li>
+          <h5>Increased disk usage</h5>
+          <p>
+            Table bloat can cause the disk space usage to increase
+            unnecessarily, which can lead to disk space shortages and degraded
+            performance.
```

Comment:
I think the last part is a little vague. And I don't think we need to equivocate about the impact: except in some corner cases, bloat will always have a negative impact (no?). Maybe something like
```suggestion
            Table bloat causes disk space usage to increase
            unnecessarily. It also leads to more I/O to read and write the
            same amount of actual data.
```

Reply:
Uhmm I kinda like the first sentence of your suggestion of "slower queries"

> Since bloated tables take more physical space for the same amount of actual data

I wonder if we can switch the order of "Increased disk usage" and "Slower queries", so that above sentence's "since" would read better. I'll try that version, let me know what you think.
Only downside is, I think query impact is more important than the system impact, so having the system impact item before the query impact might not be the ideal.

Reply:
Actually I'll keep the order, then change the title from "Increased disk usage" to "Increased disk usage and I/O" to show that it's more system oriented